lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -88,7 +89,18 @@ defmodule Trento.Domain.SapSystem do
     embeds_one :application, Application
   end
 
+  def execute(
+        %SapSystem{},
+        %RegisterDatabaseInstance{
+          system_replication: "Secondary",
+          system_replication_status: "ACTIVE"
```

Comment:
I wouldn't honestly check the status field. I would only check what the sap admin configured, which is the role, `Primary/Secondary`. 
Otherwise, we make everything much more complicated, even in the deregistration, as we would need to handle more scenarios.

Reply:
rixed